### PR TITLE
auth web: make max request/response body size configurable

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
Because some people have big zones.

Bumping the buffer size from 1024 to 16000 changes the read time for a 4 megabyte request from 2 minutes to around 10 seconds.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/webserver.cc [hunk 1]
```diff
@@ -281,13 +281,14 @@ try {
...
   yarl.initialize(&req);
+  req.max_request_size=d_maxbodysize;
   int timeout = 5;
   client->setNonBlocking();
 
   try {
     while(!req.complete) {
       int bytes;
-      char buf[1024];
+      char buf[16000];
```
Comment (rgacogne) [inline pdns/webserver.cc]: I wonder if we should make this configurable (and in that case perhaps move it to the heap?). This should be fine on most OS, though.

File: pdns/recursordist/docs/settings.rst [hunk 2]
```diff
@@ -1523,6 +1523,15 @@ IP address for the webserver to listen on.
 
 These subnets are allowed to access the webserver.
 
+.. _setting-webserver-max-bodysize:
+
+``webserver-max-bodysize``
+--------------------------
+-  Integer
```
Comment (rgacogne) [inline pdns/recursordist/docs/settings.rst]:
```suggestion
.. versionadded:: 4.2.0

-  Integer
```

Comment (Habbie) [reply pdns/recursordist/docs/settings.rst]: Oh yes, sorry. Will hold off on this a bit in case we also backport.

**Comment by zeha** [comment]
Maybe default to 20MB?

**Comment by Habbie** [comment]
Rebased without changes.

**Comment by Habbie** [comment]
> Maybe default to 20MB?

As long as auth happens after receiving the whole request, I'm more comfortable with a smaller limit.

File: pdns/webserver.cc (hunk 1 above)
Comment (Habbie) [reply pdns/webserver.cc]: At 16000 my impression is that actual processing of the request will be a way bigger part of execution time anyway, so I'm not sure we need yet another config option.

File: pdns/recursordist/docs/settings.rst (hunk 2 above)
Comment (Habbie) [reply pdns/recursordist/docs/settings.rst]: committed

**Pull request merged**